### PR TITLE
paho-mqtt-cpp: allow shared library builds on windows

### PR DESCRIPTION
**Pull request by vkes**
**paho-mqtt-cpp/1.2.0**

Removing superfluous restriction on Windows. With this PR, paho-mqtt-cpp can be built as shared library (DLL) on Windows. Fixes issue #8097.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — DISMISSED** [review]
Seems like relly old code :shrug:

File: recipes/paho-mqtt-cpp/all/conanfile.py [hunk 1]
```diff
@@ -33,13 +33,13 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
+        if self.options.shared:
+            del self.options. fPIC
```
Comment (SSE4) [inline recipes/paho-mqtt-cpp/all/conanfile.py:37]: NR space is not needed

**Review by SSE4 — DISMISSED** [review]

File: recipes/paho-mqtt-cpp/all/conanfile.py (hunk 1 above)
Comment (vkes) [reply recipes/paho-mqtt-cpp/all/conanfile.py]: Fixed.

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Comment by vkes** [comment]
> Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

Sorry for that. And thank you for the review!

**Pull request merged**